### PR TITLE
tests: move to pytest

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 0aa5efe8a879d448d7eb36856694df7efe58942e

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 21966d547e7bad43c0b2d84503eb9fd2ea9295f2

File: gnocchi/tests/functional/test_gabbi.py [hunk 1]
```diff
@@ -28,9 +32,8 @@
...
 
-def load_tests(loader, tests, pattern):
-    """Provide a TestSuite to the discovery process."""
+def pytest_generate_tests(metafunc):
     test_dir = os.path.join(os.path.dirname(__file__), TESTS_DIR)
-    return driver.build_tests(test_dir, loader, host=None, prefix=PREFIX,
-                              intercept=fixtures.setup_app,
-                              fixture_module=fixtures)
+    driver.py_test_generator(test_dir, prefix=PREFIX,
+                             intercept=fixtures.setup_app,
```
Comment (sileht) [inline gnocchi/tests/functional/test_gabbi.py]: @cdent any idea http://paste.openstack.org/show/622670/ ?

Looks like gabbi does not call fixtures.setup_app()

Comment (cdent) [reply gnocchi/tests/functional/test_gabbi.py]: Zounds, moving to pytest, that's exciting. It's supposed to do fixtures, so I'm not sure what's going on, I'll pull this down and try it myself.

**Comment by cdent** [comment]
The problem seems to be a conflict between existing tests (many of which aren't being collected properly because they have __init__ and py.test doesn't like that) and something going on in the discovery process. The result is that none of the gabbi fixtures run. I can get successful gabbi run with the following adjustment and tox call:

```diff
diff --git a/run-tests.sh b/run-tests.sh
index d7cd7ac1..617ba6dd 100755
--- a/run-tests.sh
+++ b/run-tests.sh
@@ -26,7 +26,7 @@ do
                        ./tools/pretty_tox.sh $*
                 ;;
             *)
-                pifpaf -g GNOCCHI_INDEXER_URL run $indexer -- pytest -v --ignore=gnocchi/tests/functional_live $* gnocchi/tests
+                pifpaf -g GNOCCHI_INDEXER_URL run $indexer -- pytest -v --ignore=gnocchi/tests/functional_live ${*:-gnocchi/tests}
                 ;;
         esac
         # NOTE(sileht): Start all storage tests at once
```

`tox -epy27-postgresql-file -- -svx gnocchi/tests/functional/test_gabbi.py`

but then pifpaf gets upset and exits non-zero

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 9bc72ff630717122253a1d8c0aa449c0af70e43a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 33696560fff36bead3d343af0c2042ea04bf04a5

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4ded8e1be9834bf7a4e27ba06f445fd5c358c06b

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 0ad8d971040ba0a3189634168698799c3827d92b

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 346df3bddf5eb3239b0ad9c3095e0f24e593a683

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5e4f4fd2ad0644856a7534fb5f55d64ece3fe2db

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 0937e7980329fb969e31cbcac2905cabc0e2dece

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 9d0f56f7e4d78587e3f470d26f7bccca269ae812

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 8e272e75b4221f250da98411e2ea0a4b6ca92840

**Comment by sileht** [comment]
Pull-request updated, HEAD is now f99b6753007f38f39dd9188df7b4bf0bc32f1c42

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 8b51b875bae96243a73b71d4e0e24b1323578c48

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4f792c60a681148a646dfa2c478b2c788d8dc737

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 705a2419495f8f46bba13cbcecca882a410c1de6

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 705a2419495f8f46bba13cbcecca882a410c1de6

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 87c030cfb444743e97af7db151ab9d2aaed9f275

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 434a19bed330b1363a763d69ca1a1b8c07e144f5

**Comment by sileht** [comment]
Pull-request updated, HEAD is now e3148d9eef4f210141bc808fb913d6b0c98094ab

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d9a659d541b1a2b3674e538b2cd4cc2f0ad836ea

**Pull request closed without merging**